### PR TITLE
AutotoolsDeps multi level customization

### LINKED ISSUE #9176: [feature] New AutotoolsDeps customization
The `AutotoolsDeps` generator should precalculate things in the constructor to allow the user to modify them before calling the `generate()` method.

### PR DESCRIPTION
**Pull request by lasote**

### CODE REVIEW

File: conan/tools/gnu/autotoolsdeps.py [hunk 1]
```diff
@@ -21,43 +31,84 @@ def cpp_info(self):
...
-        env.append("CXXFLAGS", cxxflags)
-        env.append("CFLAGS", cflags)
-        return env
-
-    def generate(self, env=None, auto_activate=True):
-        env = env or self.environment()
-        env.save_script("conanautotoolsdeps", auto_activate=auto_activate)
+        if not self._env:
+            env = Environment(self._conanfile)
+            env.append("CPPFLAGS", self._cpp_flags or self.cpp_flags)
```
Comment (memsharded) [inline conan/tools/gnu/autotoolsdeps.py:105]: This idiom here ``self._cpp_flags or self.cpp_flags`` with the second also computing the former is very confusing.

File: conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py [hunk 2]
```diff
@@ -85,5 +94,54 @@ def test_foo():
...
+             "compiler.libcxx": "libstdc++11",
+             "compiler.version": "7.1",
+             "cppstd": "17"})
+        deps = AutotoolsDeps(consumer)
+
+        # We can access the cpp_info and modify it
+        deps.cpp_info.includedirs = ["/other/includedir"]
+
+        # Later we can access to the flags and modify them
+        deps.cpp_flags.append("NewCppFlag")
```
Comment (memsharded) [inline conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py:126]: I am not sure that adding flags is a use case for ``AutotoolsDeps``, it should to go ``AutotoolsToolchain``.
I think that some control over the dependencies flags, like enabling/disabling/filter some or all of the dependencies flags might be useful or necessary, but adding a new flag? Probably not here?

**Review by memsharded — COMMENTED** [review]
Yes, this code is a bit difficult to know what is doing, with the ``_check_env_already_calculated`` and the laziness. A

File: conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py (hunk 2 above)
Comment (lasote) [reply conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py:126]: Well, that is a test, the cpp_flags are the ones filled from the deps information.

**Comment by lasote** [comment]
So? I totally agree the laziness is confusing, so any conclusion about what should we allow to customize?
Maybe just the `cpp_info` is enough and the more natural.

File: conan/tools/gnu/autotoolsdeps.py (hunk 1 above)
Comment (lasote) [reply conan/tools/gnu/autotoolsdeps.py:105]: Agree. Needed tricks to allow that level of customization. I went to the maximum level of customization so we can decide the level we want.

**Comment by lasote** [comment]
After discussing I'm closing this PR to open a new one only allowing to change the environment.

**Pull request closed without merging**